### DISCUSSION TITLE
Optimising developer resources

### DISCUSSION THREAD

**Discussion by reubenalfred** [Megabit/Blazorise#4832] [Ideas]
Hi Team,

TL;DR

@Blazorise has done a fantastic job in lowering the bar for developers new to web and Blazor development and I am one of those grateful ones :) It would be nice to provide the same for entry into OSS development too!

There are a lot of C# developers who can contribute to a project but may not have the necessary skills to engineer a relatively complex subsystem for a given project but may have the skillset and the heart to contribute to many simpler aspects of required features. It would be a shame not to have a process in place to encourage this class of developers, myself included :)

What I suggest is that, not sure if this would add to Blazorise developers' workload considerably, to decompose any work, perhaps experiment with low-priority feature requests, into smaller chunks that can be developed by less experienced but willing developers to contribute to. This way the low-priority work gets some traction, the developers gain valuable experience in the OSS methodology and build confidence working with OSS technology, improve familiarity with the project and so on. Furthermore, there may be tools and utilities in a project like Blazorise still required and these too are candidates for this type of development methodology.

The process I propose is to have a hierarchical decomposition of the solution and the requirement documented at the root level. Somebody else might then add any comments to disambiguate or clarify something in the requirement. The architect might add the 1st level of modules required and their functionalities. This iterative decomposition might be repeated until the minimum level of competency required to participate by a developer is reached.

The implementation stage will obviously start with the node level and lock once a volunteer is assigned to it, to prevent doubling up. once all formalities are completed in a level the next level can be unlocked for implementation and so on.

When the entry level is lowered for a project contribution there will be more developer resources available, more resources mean higher project activity which means frequent feature additions, bug fixes, and many more benefits...

I don't know how practical and beneficial this suggestion would be but I thought it was worth suggesting so we can say we have tried options to refine our processes :)

**Comment by stsrki** [comment]
I like your proposal—especially the comment about writing more details about the requirement and improving the ticket documentation. If that improves, then it could work.

In the past, we tried to bring more community members into the development process via some _hackatons_, but not everyone is willing to contribute. That is the main issue in this case.

**Reply by reubenalfred** [reply]
>> but not everyone is willing to contribute. That is the main issue in this case.

I see, and that's a shame perhaps we the community or any community that uses an OSS project need to appreciate that if it is always take and take without any consideration for giving something back or leaving it to the next person could leave the project in a vulnerable position and awareness needs to be raised.

I'll see what I can do and suggest something later tonight :)

every problem has a solution said Albert,,,, so lets stay positive!!

**Reply by reubenalfred** [reply]
First and foremost we assess how much work is involved to put it into action and doesn't depend on Blazorise core developers to set it up as they are probably overloaded.

Then we decide on a target audience, the type of developers we look for, highly skilled developers are hard to find because they are probably committed to something else, remember when Google setup their servers in the beginning they didn't use dedicated hardware components only COTS, similarly we need to find similar ways to fill our needs.

Once we have agreed on the class of developers we want to bring in then we set a strategy to attract them.

We provide tutorial content for that target audience to up their skills by using Blazorise to explain concepts and also videos to highlight the benefits of contributing to open source projects and the required skills to work in OSS projects.

With AI assisted content creation technology it is easy to create videos for YouTube. These are instructional and fact based videos hence doesn't have to have a human presenter.

There are developers willing to contribute to projects but we need to show them how it is done.

I'll tell you a story. About 300 years ago when pineapple was introduced in the UK it didn't catch on but when it was reintroduced about 150 years ago it a big success. So what did they do differently the 2nd time round?

The first time the people were eating it like an apple because the name of the fruit is pine-apple.

The second time round they paid attention to show people how to eat it and it worked.

We need to take the knife to the rock than trying to bring the rock to the knife to sharpen it.
